hmac/src/optim.rs

```diff
@@ -104,57 +89,67 @@ impl<D: EagerHash> KeyInit for HmacCore<D> {
     }
 }
 
-impl<D: EagerHash> UpdateCore for HmacCore<D> {
+impl<D: EagerHash> Update for Hmac<D> {
     #[inline(always)]
-    fn update_blocks(&mut self, blocks: &[Block<Self>]) {
-        self.digest.update_blocks(blocks);
+    fn update(&mut self, data: &[u8]) {
+        self.digest.update(data);
     }
 }
 
-impl<D: EagerHash> FixedOutputCore for HmacCore<D> {
+impl<D: EagerHash> FixedOutput for Hmac<D> {
```

Comment:
I'm not exactly happy with this change, I think this removes some amount of optimization as we can't access the inner `FixedOutputCore`

cc @newpavlov

Reply:
It's why we have the `CoreProxy` trait. For the optimized implementation you should add bounds on `D::Core` and use it in the generic implementation. We may need to add a `from_core` method to it (which previously was an inherent method on `CoreWrapper`). I planned to do it a bit later.

Reply:
I missed `CoreProxy`!